### PR TITLE
Library naming and man pages

### LINKED ISSUE #3345: wlfreerdp does not have a man page
The Wayland FreeRDP client is probably very identical to the X11 FreeRDP client regarding information provided on their man pages.

Currently, the Wayland FreeRDP client does not have a man page. My recommendation is to ship one man page for both packages and symlink wlfreerdp -> xfreerdp. The man page would only need minor changes, e.g. in the NAME field, the SYNOPSIS field,  the DESCRIPTION field and maybe at a few other locations.

Those changes should mention both application names, e.g.:

```
SYNOPSIS
       xfreerdp [file] [options] [/v:server[:port]]
```

->

```
SYNOPSIS
       xfreerdp [file] [options] [/v:server[:port]]
       wlfreerdp [file] [options] [/v:server[:port]]
```

For Debian, the policy prescribes to ship man pages for every executable ending up in a package. At best, those man pages are provided by upstream.

### PR DESCRIPTION
**Pull request by bmiklautz**
This pull request covers multiple issues and problems when building packages:
* the major version is now included in the library name to allow multiple parallel installations
* missing man pages where added

Besides those main points it also contains some small improvements and fixes:

* shadow server: add more error messages
* makecert: fix a regression with key lengths  > 2048 bit
* winpr-makecert: use sha256
* utils: remove unnecessary debug message

Fixes #3345 #3346 #3460 #3347

### CODE REVIEW

**Comment by bmiklautz** [comment]
@freerdp-bot test.

File: server/shadow/freerdp-shadow-cli.1.in [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+[\fB/sam-file:\fP\fI<file>\fP]
+[\fB/version\fP]
+[\fB/help\fP]
+.SH DESCRIPTION
+.B freerdp\-shadow\-cli
+can be used to share a running X display like with VNC but by using the RDP
+instead. It is also possibly to share only parts (rect) of the display.
+.SH OPTIONS
+.IP /ipc-socket:<icp-socket>
+If this option is set an icp socket with the path \fIicp-socket\fP is used
```
Comment (mfleisz) [inline server/shadow/freerdp-shadow-cli.1.in]: Typo - should be ipc

File: server/shadow/freerdp-shadow-cli.1.in [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+instead of a TCP socket.
+.IP /port:<port>
+Set the port to use. Default is 3389.
+This option is ignored if icp-socket is used.
```
Comment (mfleisz) [inline server/shadow/freerdp-shadow-cli.1.in]: Typo - ipc

File: winpr/tools/hash-cli/winpr-hash.1.in [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+.SH SYNOPSIS
+.B winpr-hash
+\fB-u\fP username
+\fB-p\fP password
+[\fB-d\fP domain]
+[\fB-f\fP { \fIdefault\fP | sam }]
+[\fB-v\fP { \fI1\fP | 2 }]
+.SH DESCRIPTION
+.B winpr-hash
+is a small utility that can be used to create a NTLM hash form a username and password pair. The created hash can be outputed as plain hash or in SAM format.
```
Comment (mfleisz) [inline winpr/tools/hash-cli/winpr-hash.1.in]: Typo: form -> from

File: winpr/tools/makecert-cli/winpr-makecert.1.in [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+[\fB-path\fP outputpath] 
+[outputname]
+.SH DESCRIPTION
+.B winpr-makecert
+is a tool for generating X.509 certificates modeled after the Windows command
+MakeCert. winpr-makecert aims to be command line compatible with MakeCert
+however not all options are supported or implemented yet.
+
+Unimplemented features are not described here. They are marked as "Unsupported"
+in in winpr-makecerts help.
```
Comment (mfleisz) [inline winpr/tools/makecert-cli/winpr-makecert.1.in]: double "in"

**Pull request merged**